ci/build_wheel_python.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="cudf"
+package_dir="python/cudf"
+
+export SKBUILD_CMAKE_ARGS="-DUSE_LIBARROW_FROM_PYARROW=ON"
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+# Downloads libcudf wheel from this current build, then points pip to it in PIP_FIND_LINKS below
+RAPIDS_PY_WHEEL_NAME="libcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp /tmp/libcudf_dist
+
+export PIP_FIND_LINKS="/tmp/libcudf_dist"
```

Comment:
```suggestion
echo "libcudf-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo /tmp/libcudf_dist/libcudf_*.whl)" > ./constraints.txt
export PIP_CONSTRAINT="${PWD}/constraints.txt"
```

instead of `PIP_FIND_LINKS` for this purpose, could you please switch to the approach from https://github.com/rapidsai/build-planning/issues/69#issuecomment-2174556118?

I found that to be a more reliable way than `--find-links` to ensure that CI fails if the wheel you just downloaded is broken.